### PR TITLE
Use custom Hugging Face models

### PR DESCRIPTION
**Pull request by geoand**
The reason this done is that
we have to support Hugging Face
models that run on different hosts
and that don't support all the
required by Langchain4j parameters

Supercedes: #38

### CODE REVIEW

File: hugging-face/runtime/src/main/java/io/quarkiverse/langchain4j/huggingface/QuarkusHuggingFaceChatModel.java [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+    public Response<AiMessage> generate(List<ChatMessage> messages) {
+
+        TextGenerationRequest request = TextGenerationRequest.builder()
+                .inputs(messages.stream()
+                        .map(ChatMessage::text)
+                        .collect(joining("\n")))
+                .parameters(Parameters.builder()
+                        .temperature(temperature)
+                        .maxNewTokens(maxNewTokens)
+                        .returnFullText(returnFullText)
```
Comment (cescoffier) [inline hugging-face/runtime/src/main/java/io/quarkiverse/langchain4j/huggingface/QuarkusHuggingFaceChatModel.java:73]: Only if explicitly set.

File: hugging-face/runtime/src/main/java/io/quarkiverse/langchain4j/huggingface/QuarkusHuggingFaceChatModel.java [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+    @Override
+    public Response<AiMessage> generate(List<ChatMessage> messages) {
+
+        TextGenerationRequest request = TextGenerationRequest.builder()
+                .inputs(messages.stream()
+                        .map(ChatMessage::text)
+                        .collect(joining("\n")))
+                .parameters(Parameters.builder()
+                        .temperature(temperature)
+                        .maxNewTokens(maxNewTokens)
```
Comment (cescoffier) [inline hugging-face/runtime/src/main/java/io/quarkiverse/langchain4j/huggingface/QuarkusHuggingFaceChatModel.java:72]: Only if explicitly set

**Review by cescoffier — CHANGES_REQUESTED** [review]

File: hugging-face/runtime/src/main/java/io/quarkiverse/langchain4j/huggingface/QuarkusHuggingFaceChatModel.java (hunk 2 above)
Comment (geoand) [reply hugging-face/runtime/src/main/java/io/quarkiverse/langchain4j/huggingface/QuarkusHuggingFaceChatModel.java:72]: Same comment as above

File: hugging-face/runtime/src/main/java/io/quarkiverse/langchain4j/huggingface/QuarkusHuggingFaceChatModel.java (hunk 1 above)
Comment (geoand) [reply hugging-face/runtime/src/main/java/io/quarkiverse/langchain4j/huggingface/QuarkusHuggingFaceChatModel.java:73]: This is `null` by default (in the builder), so setting here does not make a difference.

Comment (geoand) [reply hugging-face/runtime/src/main/java/io/quarkiverse/langchain4j/huggingface/QuarkusHuggingFaceChatModel.java:73]: Furthermore, our Quarkus config does not set this either, so it truly is `null` unless the user explicitly configures it

**Comment by geoand** [comment]
@cescoffier seems like CI is failing in `ToolsRequestScopeTest`. Do you want me to have a look?

**Comment by cescoffier** [comment]
Go ahead!

**Pull request merged**